### ISSUE TITLE
Feature specs are frequently failing on Travis

### ISSUE THREAD

**Issue by hortongn** [uclibs/scholar_uc#1783]
Some feature specs seem to be failing more frequently on Travis that they used to.

In particular, `spec/features/edit_work_spec.rb:56` fails most of the time.  

See https://travis-ci.org/uclibs/scholar_uc/builds/311944121 for an example.

The specs run fine in our local environments.

**Comment by jayanthdeejay** [comment]
Initially suspected the issue was in FactoryBot gem as we recently moved to it from FactoryGirl and there was a version difference where in FactoryGirl 4.9 and FactoryBot 4.8.2. When I looked at the gem's repo online, though there is a stable version 4.9 for both bot and girl, for some reason, they haven't published FactoryBot 4.9 yet to rubygems.org. 

On the other hand, I looked at changes made to Hyrax gem with respect to Travis configuration. There were no changes in the recent past to the travis.yml file in v1.0.5 branch. Glen suggested adding in wait time at the specific place where tests are failing. After researching online, I found [this](https://docs.travis-ci.com/user/common-build-problems/#Capybara%3A-I%E2%80%99m-getting-errors-about-elements-not-being-found). Currently testing the same.

**Comment by jayanthdeejay** [comment]
After testing our repo in a completely new environment by installing all the dependencies from the scratch, found that it was not the issue. While one of the travis builds was in progress, stumbled into the following lines and thought it might have been the issue.

```shell
$ rvm use 2.4.1 --install --binary --fuzzy
Using /home/travis/.rvm/gems/ruby-2.4.1
** Updating RubyGems to the latest version for security reasons. **
** If you need an older version, you can downgrade with 'gem update --system OLD_VERSION'. **
Updating rubygems-update
Fetching: rubygems-update-2.7.3.gem (100%)
Successfully installed rubygems-update-2.7.3
Installing RubyGems 2.7.3
Bundler 1.16.0 installed
RubyGems 2.7.3 installed
Regenerating binstubs
RubyGems system software updated
```

Because it was really tough to understand why the tests were failing without much interaction with Travis CI, after researching online, found out that it is possible to request "Debug Build" for Travis CI and it is enabled for Scholar@UC repo. This option is available for private repos by default and is provided on request by public repos. Even after getting access to the shell to run the specs and see what exactly is wrong, it was tough to pinpoint as each debug session has a timeout of 30 mins and one will lose access and also all the save_page's and everything will be lost and need to start from scratch again. 

After going back and forth with this issue, finally was able to pinpoint what was going on. 

https://github.com/uclibs/scholar_uc/blob/501acae34741b4db9c2167d85e999fae47d8424c/spec/features/edit_work_spec.rb#L49-L51

Because it was taking Fedora to create proxy user and the build was so fast that it went further and logged out the current user before the application could create a proxy user in line 49. Adding a sec delay there resolved the issue.
